functions/src/definitions/api/interfaces.ts

```diff
@@ -25,4 +25,64 @@ interface createChecker {
   lastVotedTimestamp?: null
 }
 
-export { createVoteRequest, updateVoteRequest, createChecker }
+interface Checker {
+  name: string
+  type: "human" | "ai"
+  isActive: boolean | null
+  pendingVoteCount: number
+  last30days: last30DaysStats
+  achievements: Achievements | null
+  level: number
+  experience: number
+}
+
+interface VoteSummary {
+  category: string | null
+  truthScore: number | null
+  type: "image" | "text"
+  createdTimestamp: Date
+  votedTimestamp: Date
+  text: string | null //only for type text
+  caption: string | null //only for type image
+  needsReview: boolean //if the vote differs from the majority
+  isAssessed: boolean //if the message is assessed
+  firestorePath: string
+}
+
+interface Vote {
+  type: "image" | "text"
+  text: string | null //only for type text
+  caption: string | null  //only for type image
+  storageBucketUrl: string | null //only for type image
+  category: string | null
+  truthScore: number | null
+  isAssessed: boolean //if the message is assessed
+  finalStats: AssessedInfo | null
+}
+
+interface last30DaysStats {
+  totalVoted: number
+  accuracyRate: number
+  averageResponseTime: number
+  peopleHelped: number
+}
+
+interface Achievements {
+  //TODO: add more achievements in future
+}
+
```

Comment:
Is this a valid interface as well?

```suggestion
export type VoteRequest = {
  factCheckerDocRef: DocumentReference
  platformId: string | null
  platform: "whatsapp" | "telegram" | "agent"
  hasAgreed: boolean | null
  triggerL2Vote: boolean | null
  triggerL2Others: boolean | null
  sentMessageId: string | null
  vote: 0 | 1 | 2 | 3 | 4 | 5 | null
  category: string | null
  createdTimestamp: Timestamp | null
  acceptedTimestamp: Timestamp | null
  votedTimestamp: Timestamp | null
}
```

Reply:
For the purposes of the webapp, the API will expose a "vote" interface that is a combination of voteRequest and message. Basically for this interface I'm coming from the perspective of the checkers app, not so much the database schema.